src/core/annotation.js

```diff
@@ -1367,6 +1375,71 @@ class WidgetAnnotation extends Annotation {
     }
     return localResources || Dict.empty;
   }
+
+  _collectJS(entry, xref, list, parents) {
+    if (!entry) {
+      return;
+    }
+
+    let parent = null;
+    if (isRef(entry)) {
+      if (parents.has(entry)) {
+        // If we've already met entry then we've a cycle
```

Comment:
Nit: `met` -> `found` sounds better, and please add a period at the end of the sentence.